motionplan/plan.go

```diff
@@ -396,8 +396,12 @@ func (p PlanState) Serialize() map[string]interface{} {
 	for fName, conf := range p.configuration {
 		confMap[fName] = referenceframe.InputsToFloats(conf)
 	}
-	m["poses"] = poseMap
-	m["configuration"] = confMap
+	if p.poses != nil {
```

Comment:
why this is necessary is not immediately obvious to me.
if `p.poses == nil` then `m["poses"] = posemap === m["poses"] = nil` so in my mind was harmless before.

Reply:
it's the difference between adding nil under m["poses"] and not having any "poses" key in m at all